code-challenges.test.js

```diff
@@ -15,40 +15,138 @@
 
 // a) Create a test with an expect statement using the variable provided. HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-var colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+describe("Shuffles", () => {
 
+    var colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+
+
+        it("removes the first item from the array and shuffles the remaining content", () => {
+
+
+
+          expect(Shuffles(colors1)).toEqual(["yellow", "blue", "pink", "green"])
+          expect(Shuffles(colors2)).toEqual(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"])
+
+
+        })
+      })
 
 // b) Create the function that makes the test pass.
 
+function shuffle(array) {
+    let currentIndex = array.length,  randomIndex;
+
+    // While there remain elements to shuffle...
+    while (currentIndex != 0) {
+
+      // Pick a remaining element...
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex--;
+
+      // And swap it with the current element.
+      [array[currentIndex], array[randomIndex]] = [
+        array[randomIndex], array[currentIndex]];
+    }
+
+    return array;
+  }
+
+  var colors1 = ["purple", "blue", "green", "yellow", "pink"]
+  shuffle(colors1);
+//   console.log(colors1);
+
 
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
+
+describe("numbers", () => {
+
+    var nums1 = [3, 56, 90, -8, 0, 23, 6]
+    var nums2 = [109, 5, 9, 67, 8, 24]
+
+        it("takes an array of numbers and returns an array of the minimum and maximum numbers", () => {
+
+
+
+          expect(numbers(nums1)).toEqual([-8, 90])
+          expect(numbers(nums2)).toEqual([5, 109])
+
+
+        })
+      })
+
+
 
 
 
 // b) Create the function that makes the test pass.
 
+var arr = [2, 3, 5, 10, 2, -9, 3];
+
+alert("Max value is " + arrayMaxMin(arr, "Max"));
+alert("Min value is " + arrayMaxMin(arr, "Min"));
+
+function arrayMaxMin(array, selector) {
+
```

Comment:
Your test looks great, but your function doesn't match the test structure. In the test you named your function `numbers` and it took in an array as an argument. Here you have a function called `arrayMaxMin` that takes two parameters.